single panel selection toggle

solc42 opened the pull request:
Sometimes working on 13"/11" laptops with tiled apllictaion layout find myself wanting to switch mucommander to single panel cause some names are unreadable. I know that I can move separator but it is no so handy : drag left then drag right etc.  Did not find such functional already existed so implemented. It even saves previous separation ratio =)

Menu path:  "View" -> "Toggle single panel view"

To illustrate problem here are to pictures for illustration on 13". on 11 its even harder sometimes =)

![two_panel](https://cloud.githubusercontent.com/assets/19550062/15503420/79a983f6-21c2-11e6-93b0-eedc84d5bf5e.jpg)
![one_panel](https://cloud.githubusercontent.com/assets/19550062/15503421/79ae591c-21c2-11e6-9cf7-6d5c250e6fcf.jpg)

ahadas commented on src/main/java/com/mucommander/ui/action/ActionManager.java:
```diff
@@ -207,7 +207,8 @@ public static void registerActions() {
     	registerAction(new SplitVerticallyAction.Descriptor(),              new SplitVerticallyAction.Factory());
-    	registerAction(new StopAction.Descriptor(),              			new StopAction.Factory());
+		registerAction(new ToggleSinglePanelAction.Descriptor(), 			new ToggleSinglePanelAction.Factory());
+		registerAction(new StopAction.Descriptor(),              			new StopAction.Factory());
```
please fix the alignment

ahadas commented on src/main/java/com/mucommander/ui/action/impl/ToggleSinglePanelAction.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    private void showOtherPanel() {
+        mainFrame.getRightPanel().setVisible(true);
```
why is it always the right panel? don't we want to determine it according to the selected panel?

ahadas commented on src/main/java/com/mucommander/ui/action/impl/ToggleSinglePanelAction.java:
```diff
@@ -0,0 +1,102 @@
+        boolean isSinglePanelViewNow = mainFrame.toggleSinglePanel();
+
+        if (isSinglePanelViewNow) {
+            this.previousRatio = mainFrame.getSplitPane().getSplitRatio();
```
minor style issue - please don't explicitly state 'this' when not needed

ahadas commented on src/main/java/com/mucommander/ui/action/impl/ToggleSinglePanelAction.java:
```diff
@@ -0,0 +1,102 @@
+            hideOtherPanel();
+
+        } else {
+            mainFrame.getSplitPane().setSplitRatio(this.previousRatio);
```
same here

ahadas commented on src/main/java/com/mucommander/ui/main/MainFrame.java:
```diff
@@ -103,6 +103,9 @@
     private boolean foregroundActive;
 
+    /** Is single panel view? */
+    private boolean isSinglePanel;
```
another minor styling things - please rename to 'singlePanel'

ahadas commented on src/main/java/com/mucommander/ui/main/menu/MainMenuBar.java:
```diff
@@ -318,6 +321,7 @@ public MainMenuBar(MainFrame mainFrame) {
         viewMenu.add(columnsMenu);
 
         toggleToggleAutoSizeItem = MenuToolkit.addCheckBoxMenuItem(viewMenu, ActionManager.getActionInstance(ToggleAutoSizeAction.Descriptor.ACTION_ID, mainFrame), menuItemMnemonicHelper);
+        toggleSinglePanel = MenuToolkit.addCheckBoxMenuItem(viewMenu, ActionManager.getActionInstance(ToggleSinglePanelAction.Descriptor.ACTION_ID, mainFrame), menuItemMnemonicHelper);
```
I think this one should be located below the "Show tree view" option

ahadas commented:
Hi,
Thanks for posting this.
Besides the comments inside, do we want to disable operations that use the 'other' panel in single-panel view mode (like copy or move) ?

solc42 commented:
Hi.
Fixed styling.
Great idea hiding inactive - implemented =)
Moved menu position.

About disabling options: for me it is more desirable to keep them enabled. I just "enlarge" my panel of view(for better visibility) and can move/copy files to neighbor panel. It will lose a bit of productivity it operations like copy/move will be disabled.

ahadas commented:
Cool, looks great except for the typo in "hide inactive panel not always righ one" (s/righ/right)
Could you please squash these commit to a single commit?

solc42 commented:
squashed o7

ahadas commented:
Thanks again :)
